### PR TITLE
mark `Lockable` as exported and update HISTORY.md for Lockable not being exported in v1.11

### PR DESCRIPTION
**Pull request by ericphanson**
companion to https://github.com/JuliaLang/julia/pull/54612. As mentioned there, in https://github.com/JuliaLang/julia/pull/54590 we originally intended to export Lockable. This does so for v1.12, and also updates the HISTORY.md for the fact that Lockable was not exported in v1.11 to match #54612.

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ Multi-threading changes
 -----------------------
 
 * `Threads.@threads` now supports the `:greedy` scheduler, intended for non-uniform workloads ([#52096]).
-* A new exported struct `Lockable{T, L<:AbstractLock}` makes it easy to bundle a resource and its lock together ([#52898]).
+* A new unexported struct `Lockable{T, L<:AbstractLock}` makes it easy to bundle a resource and its lock together ([#52898]).
```
Comment (ericphanson) [inline HISTORY.md:76]:
```suggestion
* A new public (but unexported) struct `Base.Lockable{T, L<:AbstractLock}` makes it easy to bundle a resource and its lock together ([#52898]).
```

**Comment by ericphanson** [comment]
test failures seem unrelated

**Comment by ericphanson** [comment]
bump

**Comment by ericphanson** [comment]
Bump

**Review by IanButterworth — APPROVED** [review]

**Pull request merged**